dv-export/python/query-changes/README.md

```diff
@@ -10,25 +10,160 @@ The application performs the following operations:
+- `-b 7000005 -d 7000006 -m restate -q queries/employee.json`: Fetches the full history (restate) for the Employee if the EmployeeID is among the changed values between data versions `7000005` and `7000006`.
+- `-e 92edafd0-22aa-11ef-9027-6fa2c7d9d5d2 -m last -q queries/productivity.json`: Fetch the last changes for the `Productivity` analytic object if the `EmployeeID` is among the changed values between data versions, using the existing export with the UUID `92edafd0-22aa-11ef-9027-6fa2c7d9d5d2`.
+- `-e 92edafd0-22aa-11ef-9027-6fa2c7d9d5d2 -m restate -q queries`: Fetch the full history (restate) for all queries in the `queries` directory using the existing export with the UUID `92edafd0-22aa-11ef-9027-6fa2c7d9d5d2`.
+
+## Data Samples:
+
+Let's consider the timeline for analytic object `Employee` and the `DataVersion` as follows.
+
+`V#`: These labels represent different versions of the Employee analytic object,
+indicating changes that have occurred over time.
+
+`DV#`: These labels represent different data versions in the Visier project.
+
+```plaintext
+Employee1:   ------V1--V2-------------------------V3------V4------->
+Employee2:   ---------------------V1---V2-------------V3----------->
+DataVersion: -DV1-----------DV2------------DV3-----------------DV4->
+```
+
+The same data is presented in a table format with dates:
+
+| Date       | Employee1 | Employee2 | DataVersion |
+|------------|-----------|-----------|-------------|
+| 2024-01-01 |           |           |             |
+| 2024-01-05 |           |           | DV1         |
+| 2024-01-10 | V1        |           |             |
+| 2024-01-15 | V2        |           |             |
+| 2024-01-25 |           |           | DV2         |
+| 2024-02-01 |           | V1        |             |
+| 2024-02-05 |           | V2        |             |
+| 2024-02-10 |           |           | DV3         |
+| 2024-02-20 | V3        |           |             |
+| 2024-03-01 |           | V3        |             |
+| 2024-03-10 | V4        |           |             |
+| 2024-03-25 |           |           | DV4         |
+
+### Restate Mode Example:
+
+The `restate` mode fetches the full history for an analytic object that has changed between two data versions.
+The results will vary depending on when the application is run.
+
+#### Example 1: Running the Application on 2024-01-26
+
+The data at this date looks like:
+
+```plaintext
+Employee1:   ------V1--V2------->
+Employee2:   ------------------->
+DataVersion: -DV1-----------DV2->
+```
+
+Command: `-b DV1 -d DV2 -m restate -q queries/employee.json`
+During this period, only `Employee1` has changes. The resulting table will look like:
+
+| Date       | Employee1 |
+|------------|-----------|
+| 2024-01-10 | V1        |
+| 2024-01-15 | V2        |
+
+#### Example 2: Running the Application on 2024-03-26
```

Comment:
This could be made clearer. That it's called with DV3 and DV4, yields E1 and E2 having changed and the subsequent `list` query will return all changes.

Reply:
To make it shorter and cleaner I added Employee3, so now we have Employee1,2,3 and DV1-DV3.
I left only two samples, to make it easier to go through.